tests/test_lazyflow/test_operators/test_ioOperators/testOpH5N5WriterBigDataset.py

```diff
@@ -0,0 +1,197 @@
+
+        h5_opWriter = OpH5N5WriterBigDataset(graph=graph)
+        n5_opWriter = OpH5N5WriterBigDataset(graph=graph)
+
+        # This checks that you can give a preexisting group as the file
+        h5_g = hdf5File.create_group("volume")
+        n5_g = n5File.create_group("volume")
+        h5_opWriter.h5N5File.setValue(h5_g)
+        n5_opWriter.h5N5File.setValue(n5_g)
+        h5_opWriter.h5N5Path.setValue("data")
+        n5_opWriter.h5N5Path.setValue("data")
+        h5_opWriter.Image.connect(opPiper.Output)
+        n5_opWriter.Image.connect(opPiper.Output)
+
+        # Force the operator to execute by asking for the output (a bool)
+        h5_success = h5_opWriter.WriteImage.value
+        n5_success = n5_opWriter.WriteImage.value
+        assert h5_success
+        assert n5_success
+
+        hdf5File.close()
+        n5File.close()
+
+        # Check the file.
+        hdf5File = h5py.File(h5_path, "r")
+        n5File = z5py.N5File(n5_path, "r")
+        h5_dataset = hdf5File[datasetInternalPath]
+        n5_dataset = n5File[datasetInternalPath]
+        assert h5_dataset.shape == dataShape
+        assert n5_dataset.shape == dataShape
+        assert (numpy.all(h5_dataset[...] == testData.view(numpy.ndarray)[...])).all()
+        assert (numpy.all(n5_dataset[...] == testData.view(numpy.ndarray)[...])).all()
+        hdf5File.close()
+        n5File.close()
+
+    def test_Writer_3(self, graph, testData, dataShape, h5_path, n5_path, datasetInternalPath):
+        # Create the h5 file
+        hdf5File = h5py.File(h5_path)
+        n5File = z5py.N5File(n5_path)
+
+        opPiper = OpArrayPiper(graph=graph)
+        opPiper.Input.setValue(testData)
+
+        # Force extra metadata onto the output
+        opPiper.Output.meta.ideal_blockshape = (1, 1, 0, 0, 1)
+        # Pretend the RAM usage will be really high to force lots of tiny blocks
+        opPiper.Output.meta.ram_usage_per_requested_pixel = 1000000.0
+
+        h5_opWriter = OpH5N5WriterBigDataset(graph=graph)
+        n5_opWriter = OpH5N5WriterBigDataset(graph=graph)
+
+        # This checks that you can give a preexisting group as the file
+        h5_g = hdf5File.create_group("volume")
+        n5_g = n5File.create_group("volume")
+        h5_opWriter.h5N5File.setValue(h5_g)
+        n5_opWriter.h5N5File.setValue(n5_g)
+        h5_opWriter.h5N5Path.setValue("data")
+        n5_opWriter.h5N5Path.setValue("data")
+        h5_opWriter.Image.connect(opPiper.Output)
+        n5_opWriter.Image.connect(opPiper.Output)
+
+        # Force the operator to execute by asking for the output (a bool)
+        h5_success = h5_opWriter.WriteImage.value
+        n5_success = n5_opWriter.WriteImage.value
+        assert h5_success
+        assert n5_success
+
+        hdf5File.close()
+        n5File.close()
+
+        # Check the file.
+        hdf5File = h5py.File(h5_path, "r")
+        n5File = z5py.File(n5_path, "r")
+        h5_dataset = hdf5File[datasetInternalPath]
```

Comment:
Does writing "lots of tiny blocks" also mean that the block size within the H5 (or N5) should reflect that size?
If we have any expectations from the stored block ize, than maybe we could if `dataset.chunks` match the shape we're expecting

Reply:
I suppose this is a valid idea worth investigating, but this file is mostly the same as [original](https://github.com/ilastik/lazyflow/blob/master/tests/testOpH5N5WriterBigDataset.py) so I don't plan to introduce any changes to it.

Reply:
Reverted changes to this test, they were made during investigating reason behind zero division.